kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -143,8 +143,11 @@ private static class ChangeDataFeedFeature extends TableFeature.LegacyWriterFeat
 
     @Override
     public boolean hasKernelWriteSupport(Metadata metadata) {
-      // writable if change data feed is disabled
-      return !TableConfig.CHANGE_DATA_FEED_ENABLED.fromMetadata(metadata);
+      // Kernel now supports writes to CDF-enabled tables with restrictions.
```

Comment:
You don't need to block this here, this is already handled in the IceberWriterCompatXMetadataValidatorAndUpdater.

This can just return true. The comment here explaining the limited support is helpful thanks!